### PR TITLE
[alpha.webkit.UncountedCallArgsChecker] Support more trivial expressions.

### PR DESCRIPTION
**Pull request by rniwa**
Treat a compound operator such as |=, array subscription, sizeof, and non-type template parameter as trivial so long as subexpressions are also trivial.

Also treat true/false boolean literal as trivial.

### CODE REVIEW

File: clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp [hunk 1]
```diff
@@ -360,6 +370,16 @@ class TrivialFunctionAnalysisVisitor
     return TrivialFunctionAnalysis::isTrivialImpl(Callee, Cache);
   }
 
+  bool
+  VisitSubstNonTypeTemplateParmExpr(const SubstNonTypeTemplateParmExpr *E) {
+    // Non-type template paramter is trivial if the replacement is trivial.
+    return Visit(E->getReplacement());
```
Comment (haoNoQ) [inline clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp:376]: These might be always trivial? They're compile-time constants after all.

**Review by haoNoQ — APPROVED** [review]
LGTM other than the other tiny comment.

File: clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp (hunk 1 above)
Comment (rniwa) [reply clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp:376]: oh, okay. I guess we can just return true then.

**Comment by rniwa** [comment]
Thanks for the review!

**Pull request merged**